File: thaisummit/thaisummit/report/shift_assignment_report/shift_assignment_report.py
# Copyright (c) 2013, TEAMPRO and contributors
# For license information, please see license.txt

import frappe
from frappe import msgprint, _
from frappe.utils import formatdate

def execute(filters=None):
	columns = get_columns(filters)
	data = get_data(filters)
	return columns, data

def get_columns(filters):
	columns = []
	columns += [
		_("Employee ID") + ":Data/:150",_("Employee Name") + ":Data/:200",_("Department") + ":Data/:150",_("Employee Type") + ":Data/:150",_("Shift Type") + ":Data/:100",_("Shift Date") + ":Data/:150",_("Route No") + ":Data/:100",_("Boarding Point") + ":Data/:150"
	]
	return columns

def get_data(filters):
	data = []
	if filters.department:
		shifts = frappe.db.sql("""select * from `tabShift Assignment` where start_date between '%s' and '%s' and department = '%s' and docstatus = '1' """%(filters.from_date,filters.to_date,filters.department),as_dict=True)
		for s in shifts:
			data.append([s.employee,s.employee_name,s.department,s.employee_type,s.shift_type,formatdate(s.start_date),s.route_no,s.boarding_point])
	else:
		shifts = frappe.db.sql("""select * from `tabShift Assignment` where start_date between '%s' and '%s' and docstatus = '1' """%(filters.from_date,filters.to_date),as_dict=True)
		for s in shifts:
			data.append([s.employee,s.employee_name,s.department,s.employee_type,s.shift_type,formatdate(s.start_date),s.route_no,s.boarding_point])
	return data

File: thaisummit/thaisummit/report/shift_assignment_report/shift_assignment_report.js
// Copyright (c) 2016, TEAMPRO and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Shift Assignment Report"] = {
	"filters": [
		{
			"fieldname": "from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"reqd": 1,
			"default": frappe.datetime.now_date()
		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"reqd": 1,
			"default": frappe.datetime.now_date()
		},
		{
			"fieldname": "department",
			"label": __("Department"),
			"fieldtype": "Link",
			"options": "Department"
		}
	],

	// onload: function (report) {
	// 	var to_date = frappe.query_report.get_filter('to_date');
	// 	to_date.refresh();
	// 	to_date.set_input(frappe.datetime.add_days(frappe.datetime.month_start(), 24))

	// 	var from_date = frappe.query_report.get_filter('from_date');
	// 	from_date.refresh();
	// 	var d = frappe.datetime.add_months(frappe.datetime.month_start(), -1)
	// 	from_date.set_input(frappe.datetime.add_days(d, 25))

	// 	var employee = frappe.query_report.get_filter('employee');
	// 	employee.refresh();
	// 	var employee_type = frappe.query_report.get_filter('employee_type');
	// 	employee_type.refresh();
	// 	if (frappe.session.user != "Administrator") {
	// 		frappe.db.get_value("Employee", { 'user_id': frappe.session.user }, ["name", "employee_type"], (r) => {
	// 			employee.set_input(r.name)
	// 			employee_type.set_input(r.employee_type)
	// 			report.refresh()
	// 		})
	// 	}
	// },
};
